Wikipedia/Code/MWKSearchResult.m

```diff
@@ -235,4 +235,33 @@ - (nullable NSURL *)articleURLForSiteURL:(NSURL *)siteURL {
     return [siteURL wmf_URLWithTitle:self.displayTitle];
 }
 
+#pragma mark - NSObject
+
+- (BOOL)isEqualToSearchResult:(MWKSearchResult *)searchResult {
+    if (!searchResult) {
+        return NO;
+    }
+    
+    BOOL haveEqualDisplayTitles = (!self.displayTitle && !searchResult.displayTitle) || [self.displayTitle isEqualToString:searchResult.displayTitle];
+    
+    return haveEqualDisplayTitles;
+}
+
+- (BOOL)isEqual:(id)object {
+    if (self == object) {
+        return YES;
+    }
+    
+    if (![object isKindOfClass:[MWKSearchResult class]]) {
+        return NO;
+    }
+    
+    return [self isEqualToSearchResult:(MWKSearchResult *)object];
+    
+}
+
+- (NSUInteger)hash {
+    return [self.displayTitle hash];
+}
```

Comment:
Thanks @julienbodet, I'm not 100% confident about the solution, my understanding is that `containsObject` returns `false` cause those objects are equal, but not identical.